[WIP] New documentation site

ndelangen opened the pull request:
Issue: -documentation system is limited and hard to maintain-

## What I did
I build a new documentation site built with next.js

## Goals
- [x] pages based on markdown & react
- [x] great multi-framework support ( progress by @ndelangen )
- [x] pages should be fully viewable in GitHub 
- [x] great dev experience (HMR)
- [x] great user experience (rich design)
- [ ] great accessibility
- [ ] fast, first paint and TTI
- [x] automatic contributions ( progress by @ndelangen )
- [x] multiple versions support ( progress by @igor-dv )
- [ ] deploy from CI to main URL when on master
- [x] deploy to branch-URL when on PR ( handled by netlify )
- [x] Place branch-URL on PR 
- [ ] prepare for multi-language support
- [ ] provide a page with statistics on the repo & npm, maybe some nice graphs? ( why storybook )
- [ ] embed a storybook example
- [ ] list of community examples

## How to test

It's on netlify

ndelangen committed:
WIP

ndelangen committed:
IMPROVE sidenav content

ndelangen committed:
Merge branch 'master' into new-docs

ndelangen committed:
Merge branch 'release/3.3' into new-docs

Hypnosphi commented:
Great work! As far as I remember, you’ve deployed this to some custom domain. Can you bring it in?

Hypnosphi commented on docs-v2/.babelrc:
```diff
@@ -0,0 +1,23 @@
+    "next/babel",
+    ["env", {
+      "targets": {
+        "chrome": 52
```
Isn't the docsite supposed to work in a wider range of browsers?

Hypnosphi commented:
Is there a chance that some of those components would be useful as part of `@storybook/components`?

Hypnosphi commented:
Using close colours for background and text leads to bad readabiity =(

<img width="1053" alt="screen shot 2017-08-31 at 03 01 53" src="https://user-images.githubusercontent.com/6651625/29901461-b874e21e-8e00-11e7-9028-19bfce76bf85.png">

Hypnosphi commented:
Are those lines really needed?
<img width="272" alt="screen shot 2017-08-31 at 03 04 03" src="https://user-images.githubusercontent.com/6651625/29901470-cabf1c46-8e00-11e7-9e87-6a783a89123d.png">
<img width="274" alt="screen shot 2017-08-31 at 03 05 34" src="https://user-images.githubusercontent.com/6651625/29901471-cabef194-8e00-11e7-92a4-89cbe5001c98.png">

ndelangen committed:
REFACTOR headings

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.3' into new-docs

# Conflicts:
#	.eslintignore
#	yarn.lock

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.3' into new-docs

# Conflicts:
#	lib/cli/bin/generate.js

igor-dv committed:
Call bootstrap with node

igor-dv committed:
Remove the public dir content

igor-dv committed:
Add docs to the yarn workspaces

igor-dv committed:
Add public dir to ignore list

igor-dv committed:
Add git commands for committing a new docs version

igor-dv committed:
Move git scripts to build.js

igor-dv committed:
Override public export with new version export

igor-dv committed:
Merge branch 'release/3.3' into new-docs

igor-dv committed:
Remove env variable with token from the package.json (I'll revoke it, I promise)

igor-dv committed:
Remove public dir at the beginning + update version of the docs package.json

igor-dv committed:
Remove _next dir before copying

igor-dv committed:
Merge branch 'release/3.3' into new-docs

igor-dv committed:
Minor new line cleanup

igor-dv committed:
Fix typo

igor-dv committed:
Remove public dir - it will be ignored in #1855 - this will ease on CR.

igor-dv committed:
Merge remote-tracking branch 'origin/new-docs' into new-docs-versioning2

igor-dv committed:
Move "remark-helpers" to dependencies - linter fix

igor-dv committed:
Merge branch 'release/3.3' into new-docs

igor-dv committed:
Merge remote-tracking branch 'origin/new-docs' into new-docs-versioning2

# Conflicts:
#	docs-v2/package.json

igor-dv committed:
Delete _next dir right before copying to the outputDir

igor-dv committed:
Refactor build to chain of commands

igor-dv committed:
Use cross-env for setting NODE_ENV=production in the "next" related scripts

igor-dv committed:
:shit:, I did it again..

igor-dv committed:
Try to ignore docs in the build-packs command

igor-dv committed:
Make sitemap generation windows friendly

igor-dv committed:
Add version-aware link component

igor-dv committed:
Move everything from version _next dir to the main _next dir

igor-dv committed:
Remove ASSET_PREFIX it's not needed

igor-dv committed:
Remove verbose parameter from the git commands - there is a secret that might be printed.

igor-dv committed:
Add docs-v2 to CI

igor-dv committed:
Add git user config + temp remove CI steps

igor-dv committed:
Revert the CI jobs removal + remove the docs-v2 for now - it needs a better strategy than just publishing docs to the repo on every build in CI.

igor-dv committed:
Delete old _next files according to the stored reference

igor-dv committed:
Move promisifyProcess to a separate script and make the build more readable

igor-dv committed:
Generate versions.json

igor-dv committed:
Add comments about windows support

igor-dv committed:
Add docs-v2 to be deployed only on master

igor-dv committed:
Merge pull request #1855 from storybooks/new-docs-versioning2

New docs versioning - repo proposal

ndelangen committed:
Merge branch 'release/3.3' into new-docs

ndelangen committed:
Merge branch 'release/3.3' into new-docs

dangreenisrael committed:
Updated `guides/why`
- Added a value prop
- Gave an overview of the preview window
- Added a list of build in addons

Began adding content to development folder
- Setting up a dev environment
- Avoiding burnout

Hypnosphi commented on docs-v2/content/development/maintainers-avoiding-burnout.md:
```diff
@@ -0,0 +1,49 @@
+
+* * *
+
+This document has been taken **_mostly_** from <https://github.com/Storybook/brew/blob/76955b47bd2ae3eba1b041430d7e1aeab246de0d/docs/Maintainers-Avoiding-Burnout.md>.  Thanks Storybook!
```
What?

Hypnosphi commented on docs-v2/content/development/setting-up-a-dev-environment.md:
```diff
@@ -0,0 +1,106 @@
+4.  `yarn test`
+
+### 2. Create a project (skip this if you already have a react project)
+
```
Most of the time this won't be needed. Using projects under `examples` directory is the easiest way of reproducing issues. In case of React it's `examples/cra-kitchen-sink`

Hypnosphi commented on docs-v2/content/development/setting-up-a-dev-environment.md:
```diff
@@ -0,0 +1,106 @@
+# Setting up a dev environment (React)
```
Why React only?

ndelangen replied:
It shouldn't be I think.

dangreenisrael replied:
I'll change it to be setting up a dev environment, then a section for each different platform (React, ReactNative, Angular, etc)

Hypnosphi commented on docs-v2/content/development/setting-up-a-dev-environment.md:
```diff
@@ -0,0 +1,106 @@
+Please have the **_latest_** stable versions of the following on your machine
+
+-   node
+-   npm
```
`yarn` should be enough

Hypnosphi commented on docs-v2/content/development/setting-up-a-dev-environment.md:
```diff
@@ -0,0 +1,106 @@
+-   node
+-   npm
+-   yarn
+-   create-react-app
```
Not really needed, see comment for ### 2

Hypnosphi commented on docs-v2/content/development/setting-up-a-dev-environment.md:
```diff
@@ -0,0 +1,106 @@
+-   yarn
+-   create-react-app
+
+Please have at least version `3.3.0-alpha.0` of `@storybook/cli`
```
It's better to just link current version from the repo if needed

igor-dv committed:
Limit docs-v2 running on master only. "branches" field on the jobs list level doesn't work properly, so should be moved to workflows section: https://circleci.com/docs/2.0/configuration-reference/#filters

ndelangen commented on docs-v2/content/guides/why.md:
```diff
@@ -9,39 +9,66 @@ We're going to discuss the core concepts of Storybook which will list Storybook'
-:::
+## Decorators
+
+> Currently React only
```
Actually Storybook for Vue supports the api, but the decorators are different.

dangreenisrael committed:
Started setting up the contributing page of the newdocs

Hypnosphi commented on docs-v2/content/contributing/maintainers-avoiding-burnout.md:
```diff
@@ -46,4 +46,4 @@ work to be done.
 * * *
 
-This document has been taken **_mostly_** from <https://github.com/Storybook/brew/blob/76955b47bd2ae3eba1b041430d7e1aeab246de0d/docs/Maintainers-Avoiding-Burnout.md>.  Thanks Storybook!
+This document has been taken **_mostly_** from [HomeBrew](https://github.com/Storybook/brew/blob/76955b47bd2ae3eba1b041430d7e1aeab246de0d/docs/Maintainers-Avoiding-Burnout.md).  Thanks HomeBrew!
```
URL is still wrong

dangreenisrael replied:
whoops

Hypnosphi commented on docs-v2/content/contributing/release-guide.md:
```diff
@@ -0,0 +1,57 @@
+# Release Guide
```
This has been updated recently, see #1882

dangreenisrael replied:
Updated

Hypnosphi commented:
@ndelangen @dangreenisrael I wonder do we start having contributor docs in two places and keeping them in sync?

dangreenisrael committed:
Changes per @hypnosphi and @ndelangen's comments.  Setting up dev env contents still WIP

dangreenisrael commented:
Perhaps we can autogenerate a copy of the docs into /docs, and have a note at the top telling people how to get to the source files.

dangreenisrael committed:
Seperated setting-up-a-dev-environment into sections based on platform

dangreenisrael committed:
Changed ordering of setting-up-a-dev-environment.md

dangreenisrael committed:
Fixed heading issues (I think?)

dangreenisrael committed:
Added how to for kitchen sink apps

dangreenisrael committed:
Moved headers

dangreenisrael committed:
Updates as per @ndelangen and @hypnosphi's feedback on PR https://github.com/storybooks/storybook/pull/1911

dangreenisrael committed:
Merge branch 'new-docs' of github.com:storybooks/storybook into new-contributor-docs

# Conflicts:
#	docs-v2/content/contributing/setting-up-a-dev-environment.md

Commit:
Merge pull request #1886 from storybooks/new-contributor-docs

Made recommended changes

Hypnosphi committed:
Merge branch 'release/3.3' into new-docs

ndelangen committed:
Merge branch 'master' into new-docs

# Conflicts:
#	.circleci/config.yml
#	.gitignore
#	CHANGELOG.md
#	addons/actions/package.json
#	addons/centered/package.json
#	addons/comments/package.json
#	addons/events/package.json
#	addons/graphql/package.json
#	addons/info/package.json
#	addons/info/src/components/PropTable.js
#	addons/knobs/README.md
#	addons/knobs/package.json
#	addons/knobs/src/components/__tests__/Array.js
#	addons/knobs/src/components/types/Array.js
#	addons/knobs/src/index.js
#	addons/links/README.md
#	addons/links/package.json
#	addons/notes/package.json
#	addons/options/package.json
#	addons/storyshots/package.json
#	addons/storyshots/src/index.js
#	app/react-native/package.json
#	app/react-native/readme.md
#	app/react/package.json
#	app/react/src/server/index.js
#	app/vue/package.json
#	app/vue/src/server/index.js
#	docs/pages/basics/faq/index.md
#	examples/cra-kitchen-sink/.storybook/addons.js
#	examples/cra-kitchen-sink/package.json
#	examples/cra-kitchen-sink/src/stories/__snapshots__/index.storyshot
#	examples/cra-kitchen-sink/src/stories/index.js
#	examples/crna-kitchen-sink/package.json
#	examples/vue-kitchen-sink/package.json
#	examples/vue-kitchen-sink/src/stories/index.js
#	lerna.json
#	lib/addons/package.json
#	lib/channel-postmessage/package.json
#	lib/channel-websocket/package.json
#	lib/channels/package.json
#	lib/cli/bin/generate.js
#	lib/cli/generators/SFC_VUE/index.js
#	lib/cli/generators/UPDATE_PACKAGE_ORGANIZATIONS/index.js
#	lib/cli/generators/VUE/index.js
#	lib/cli/lib/helpers.js
#	lib/cli/package.json
#	lib/codemod/package.json
#	lib/components/package.json
#	lib/ui/package.json
#	lib/ui/src/modules/ui/components/addon_panel/index.test.js
#	package.json
#	scripts/test.js
#	yarn.lock

ndelangen committed:
docs-v2 in yarn workspaces

ndelangen committed:
Linting && ADD babel-plugin-preval && set polymer to "in progress"

ndelangen committed:
Linting && IMPROVE visual of sidenav

ndelangen committed:
ADD first addons docs && Linting && sitemap generation fix

ndelangen committed:
FIX contributors list by parsing page src and finding true markdown dependency, fallback on page src

ndelangen committed:
Linting

ndelangen committed:
ADD example of using the sitemap to iterate over subroutes

ndelangen committed:
ADD frameworks & videos

ndelangen committed:
MOVE icons into lib/components

ndelangen committed:
DELETE yarn.lock

ndelangen committed:
FIX Icons

ndelangen committed:
MOVE logos to lib/components

ndelangen committed:
IMPROVE guides/index && general improvements

ndelangen committed:
FIX styling of li > p

ndelangen committed:
IMPROVE styling

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	.circleci/config.yml
#	CHANGELOG.md
#	addons/actions/package.json
#	addons/centered/package.json
#	addons/comments/package.json
#	addons/events/package.json
#	addons/graphql/package.json
#	addons/info/package.json
#	addons/info/src/components/PropTable.js
#	addons/knobs/README.md
#	addons/knobs/package.json
#	addons/knobs/src/components/__tests__/Array.js
#	addons/knobs/src/components/types/Array.js
#	addons/knobs/src/index.js
#	addons/knobs/src/vue/index.js
#	addons/links/package.json
#	addons/notes/package.json
#	addons/options/package.json
#	addons/storyshots/package.json
#	addons/storyshots/src/index.js
#	addons/viewport/package.json
#	app/react-native/package.json
#	app/react/package.json
#	app/react/src/server/index.js
#	app/vue/package.json
#	app/vue/src/server/index.js
#	docs/src/pages/basics/faq/index.md
#	examples/cra-kitchen-sink/.storybook/addons.js
#	examples/cra-kitchen-sink/package.json
#	examples/cra-kitchen-sink/src/stories/__snapshots__/addon-info.stories.storyshot
#	examples/cra-kitchen-sink/src/stories/index.js
#	examples/crna-kitchen-sink/package.json
#	examples/vue-kitchen-sink/package.json
#	examples/vue-kitchen-sink/src/stories/index.js
#	lerna.json
#	lib/addons/package.json
#	lib/channel-postmessage/package.json
#	lib/channel-websocket/package.json
#	lib/channels/package.json
#	lib/cli/bin/generate.js
#	lib/cli/generators/SFC_VUE/index.js
#	lib/cli/generators/UPDATE_PACKAGE_ORGANIZATIONS/index.js
#	lib/cli/generators/VUE/index.js
#	lib/cli/lib/helpers.js
#	lib/cli/package.json
#	lib/cli/test/snapshots/meteor/package.json
#	lib/cli/test/snapshots/react/package.json
#	lib/cli/test/snapshots/react_native/package.json
#	lib/cli/test/snapshots/react_native_scripts/package.json
#	lib/cli/test/snapshots/react_project/package.json
#	lib/cli/test/snapshots/react_scripts/package.json
#	lib/cli/test/snapshots/sfc_vue/package.json
#	lib/cli/test/snapshots/update_package_organisations/package.json
#	lib/cli/test/snapshots/vue/package.json
#	lib/cli/test/snapshots/webpack_react/package.json
#	lib/codemod/package.json
#	lib/components/package.json
#	lib/ui/package.json
#	lib/ui/src/modules/ui/components/addon_panel/index.test.js
#	package.json
#	scripts/test.js
#	yarn.lock

ndelangen committed:
Merge branch 'release/3.3' into new-docs

ndelangen committed:
IMPROVE styling

ndelangen committed:
IMPROVE content pages && ADD addon docs

ndelangen committed:
Netlify for new docs

ndelangen committed:
Merge branch 'release/3.3' into new-docs

ndelangen committed:
FIX the new docs Build

Hypnosphi commented on docs-v2/components/Blocks.js:
```diff
@@ -107,12 +107,9 @@ const BlockItem = glamorous.div(
     borderRadius: '3px',
     display: 'flex',
-    transition: 'all .5s',
+    transition: 'all 2s',
```
`transition: all` is never a good idea. Please list the particular properties

ndelangen committed:
Merge branch 'release/3.3' into new-docs

ndelangen committed:
UPDATE lockfile

ndelangen committed:
FIX netlify build

ndelangen committed:
IMPROVE breadcrumb

ndelangen committed:
IMPROVE breadcrumb

ndelangen committed:
Linting

ndelangen committed:
FIX build-packs

ndelangen committed:
Merge branch 'release/3.3' into new-docs

ndelangen committed:
IMRPOVE docs addons section & styling of lists

ndelangen committed:
IMPROVE fontFamily, all from theme.js

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	addons/info/README.md
#	examples/cra-kitchen-sink/src/stories/index.stories.js
#	yarn.lock

ndelangen committed:
REFACTOR breadcrumb

ndelangen committed:
IMPROVE styling and docs

ndelangen committed:
IMPROVE many things

ndelangen committed:
IMPROVE documentation

ndelangen committed:
FIX

ndelangen committed:
CHANGE docs of addon-info to use PropTypes over Flow

ndelangen committed:
IMPROVE the docs a ton

ndelangen committed:
ADD contributor to newly added page

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	addons/info/README.md
#	package.json
#	yarn.lock

ndelangen committed:
FIX snapshots && CLEANUP

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	package.json
#	yarn.lock

ndelangen committed:
Merge branch 'release/3.3' into new-docs

ndelangen committed:
ADD writing-stories page and organisation page to docs

ndelangen committed:
FIX invalid export

ndelangen committed:
IMPROVE the writing stories section of the documentation

ndelangen committed:
UPDATE yarn.lock

ndelangen committed:
Spelling corrections and better locations for some references

ndelangen committed:
ADD organisation page

ndelangen committed:
REMOVE the docs step from CI, the validation is now handled with netlify

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	yarn.lock

ndelangen committed:
CHANGE width of content

ndelangen committed:
IMPROVE padding and alignment of sidenav items

ndelangen committed:
IMPROVE docs

ndelangen committed:
FIX netlify deploy

ndelangen committed:
REMOVE the reference to --docs && REGEN lockfile

ndelangen committed:
ADD codeswitcher based on localstorage

ndelangen committed:
IMPROVE the codeswitcher and docs

ndelangen committed:
CHANGE codeswitcher to be more accessible

ndelangen committed:
ADD contributors UI && ADD npm logo && FIX markdown styling in pageTitle

ndelangen committed:
ADD build-static-version page

ndelangen committed:
IMPROVE the build-storybook docs page, by adding package.json example

ndelangen committed:
ADD webpack guide page to docs

ndelangen committed:
ADD customization, babel page to docs

IMPROVE rendering of button
FIX required key prop warning

ndelangen committed:
IMPROVE Contributors component & REMOVE the Test component from manual install page

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	addons/info/README.md
#	examples/cra-kitchen-sink/src/stories/__snapshots__/index.stories.storyshot
#	examples/cra-kitchen-sink/src/stories/index.stories.js
#	package.json
#	yarn.lock

ndelangen committed:
UPDATE lockfile

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	yarn.lock

ndelangen committed:
FIX tests by updating to "react-split-pane": "^0.1.74",

ndelangen committed:
Merge branch 'release/3.3' into new-docs

# Conflicts:
#	yarn.lock

ndelangen committed:
FIX many proptype warnings && REFACTOR Split && reference other stories page from manual-setup

ndelangen committed:
IMPROVE the Split component to have a collapsable TOC & related-content

ndelangen committed:
FIX mobile topnav

ndelangen committed:
ADD options page to docs && ADD InlineCode markdown component && IMPROVE documentation

ndelangen committed:
IMPROVE breadcrumb

ndelangen committed:
Merge branch 'master' into new-docs

# Conflicts:
#	.circleci/config.yml
#	addons/background/README.md
#	addons/info/README.md
#	addons/links/README.md
#	addons/notes/README.md
#	addons/storyshots/README.md
#	lib/components/src/index.js
#	package.json
#	yarn.lock

ndelangen committed:
ADD demo iframe to netlify

gabrielcsapo commented:
@ndelangen is this going to be merged or are the changes stale?
